--- old/src/hotspot/cpu/x86/assembler_x86.cpp	2019-04-03 14:28:54.697451207 +0800
+++ new/src/hotspot/cpu/x86/assembler_x86.cpp	2019-04-03 14:28:54.425441860 +0800
@@ -3099,7 +3099,7 @@
     }
     return;
   }
-  if (UseAddressNop && VM_Version::is_amd()) {
+  if (UseAddressNop && (VM_Version::is_amd() || VM_Version::is_hygon())) {
     //
     // Using multi-bytes nops "0x0F 0x1F [address]" for AMD.
     //  1: 0x90
