--- old/src/hotspot/cpu/x86/vm_version_x86.cpp	2019-04-03 14:28:55.877491758 +0800
+++ new/src/hotspot/cpu/x86/vm_version_x86.cpp	2019-04-03 14:28:55.601482272 +0800
@@ -1165,7 +1165,7 @@
     }
   }
 
-  if( is_amd() ) { // AMD cpus specific settings
+  if( is_amd() || is_hygon() ) { // AMD cpus specific settings
     if( supports_sse2() && FLAG_IS_DEFAULT(UseAddressNop) ) {
       // Use it on new AMD cpus starting from Opteron.
       UseAddressNop = true;
@@ -1239,8 +1239,8 @@
     }
 #endif // COMPILER2
 
-    // Some defaults for AMD family 17h
-    if ( cpu_family() == 0x17 ) {
+    // Some defaults for AMD family 17h || Hygon family 18h
+    if ( cpu_family() == 0x17 || cpu_family() == 0x18 ) {
       // On family 17h processors use XMM and UnalignedLoadStores for Array Copy
       if (supports_sse2() && FLAG_IS_DEFAULT(UseXMMForArrayCopy)) {
         FLAG_SET_DEFAULT(UseXMMForArrayCopy, true);
